### ISSUE TITLE
USB : Guncon 2 Game compatibility report. 

### ISSUE THREAD

**Issue by Gomam0n-afk** [PCSX2/pcsx2#7619]
### GunCon 2/G-Con 2 Compatibility report (With default values only).
Tier rating system :
BAD: Game is not playable at all.
OK: Game is playable but has issues or does not work with widescreen.
PERFECT: Game is 100% Playable, with widescreen hacks/Widescreen mode turned on 

**[All testing is done with PCSX2's default scale/center/size values for the GCon2, and in HW DirectX 12 mode.]**

GunCom 2 (SLES-52620) (AKA Death Crimson OX+) : OK. 
Note :A button shoots sometimes for some reason even though it's supposed to reload only, making it difficult to play.
You also cannot calibrate it dead center at first and have to aim to the left in 1 point calibration mode.
Unrelated issues : Deinterlacing is broken in Auto mode.

Gunfighter 2 - Revenge of Jesse James (SLES-52620) : PERFECT.
Note: None.
Unrelated issue: Game title inside the db is wrong, Legend of jesse james is the 1st game on PS1, Revenge of jesse james is the 2nd.

ENDGAME (SLES-50936) : BAD.
Note: Aim overshoots left and right in 16:9, and overshoots even more top to bottom in 4:3 mode. aim is widely miscalibrated by default (more so than og hardware).
The game only responds to the 2nd USB port as player 1.
Unrelated issue: Deinterlacing is broken with FMV.

Ninja Assault (SCES-50899) : PERFECT 50Hz BAD 60Hz.
Note: Very slight overshooting in 50hz 16:9. Game is unplayable in 60hz as it overshoots widely and the gun cannot be calibrated properly.
Unrelated issues: broken deinterlacing in automatic mode.

Time Crisis 3 (SCES-51844) : OK.
Note: Game seems to calibrate perfectly but ingame your aim undershoots in all axis' in both 4:3 and 16:9.
Unrelated issues : colored overaly on top of the entire image except HUD + white square in the top left. + Broken deinterlacing.

Time Crisis 3 (Chineese) (NTSC) (SCAJ-20060) : BAD.
Note: Wide overshooting, impossible to calibrate the gun.

Virtua Cop - Elite Edition (SLES-51229) : BAD.
Note: Game seems to calibrate fine (even after a longer than usual black screen after the first shot), but ingame your aim gets stuck in the center or in an earlier position,
and moves only once every few seconds making the game entirely unplayable.

Resident Evil Code Veronica X Survivor 2 (SLES-50650) : BAD.
Note: First try at calibrating worked fine but overshot widely to the right. retried calibrating again with the calibration shot bind, 
and it inverted the Y axis and overshot my aim by a lot, restarted the game and tried calibrating again, and i got extreme overshooting on the top and bottom instead.

Crisis Zone (SCES-52530) : BAD.
Note: Cannot calibrate crosshair to the center, forced to aim right of everything, didn't try anything fruther than that.

Dino Stalker (SLES-50930) : BAD.

Note:
[  436.8596] (GunCon2) No automatic config found for 'SLES-50930'.
Calibration shots are obligatory, first shot/ some shots are detected as a double tap, i cannot aim to the very top of the screen.

**Comment by legend800** [comment]
I can confirm status on Gun Fighter 2 and Crisis Zone - the latter is probably related to your defect that you logged about cali button not working on Vampire Night. 

Correction:
- Virtua Cop - Elite Edition (SLES-51229): PERFECT
       Just double checked. It's fine in cali and in-game. You're not saving your cali settings probably. Steps: Press Start, using cali shot to set, offscreen shoot, Ok at prompt, offscreen shoot again, Ok to confirm (pain in the ass, just bad game design)

Other games:
- Time Crisis 2 (SLUS-20219): PERFECT
     Calibration and in-game are perfect (4:3 only)

- Time Crisis 3 (SLUS-20645): PERFECT
      Calibration and in-game are perfect (4:3 only)

Widescreen Patches:
All the TC games need widescreen patches disabled since the in-game shots cannot go past 4:3 area and throw off calibration: 
- TC2
- TC3
- TC Crisis Zone

**Comment by Gomam0n-afk** [comment]
> I can confirm status on Gun Fighter 2 and Crisis Zone - the latter is probably related to your defect that you logged about cali button not working on Vampire Night.
> 
> Correction:
> 
> * Virtua Cop - Elite Edition (SLES-51229): PERFECT
>   Just double checked. It's fine in cali and in-game. You're not saving your cali settings probably. Steps: Press Start, using cali shot to set, offscreen shoot, Ok at prompt, offscreen shoot again, Ok to confirm (pain in the ass, just bad game design)
> 
> Other games:
> 
> * Time Crisis 2 (SLUS-20219): PERFECT
>   Calibration and in-game are perfect (4:3 only)
> * Time Crisis 3 (SLUS-20645): PERFECT
>   Calibration and in-game are perfect (4:3 only)
> 
> ![1](https://user-images.githubusercontent.com/20983844/208335258-9ade8514-169d-468e-be18-8d0850584430.png)
> 
> Widescreen Patches: All the TC games need widescreen patches disabled since the in-game shots cannot go past 4:3 area and throw off calibration:
> 
> * TC2
> * TC3
> * TC Crisis Zone

I tried again with a brand new pcsx2 config, and Virtua Cop is indeed still completely broken.
https://streamable.com/ncm3oe

Crisis Zone and TC3 still have the exact same issue as i have described. note that i'm using PAL games **NOT** NTSC.
![image](https://user-images.githubusercontent.com/59240947/208336192-4d4f4bfc-4bdc-41fe-87c5-1b428e582b6c.png)


All games have also been tested without widescreen patches if issues were encountered. your posts is just about NTSC games.

**Comment by legend800** [comment]
Virtua Cop:
We're using the same game and it's perfect here, so it's related to how your calibrating/settings, not the emulate. It's flawless here, so try my steps again:
![2022-12-18 18_36_42-Virtua Cop - Elite Edition](https://user-images.githubusercontent.com/20983844/208336973-08cf3497-9aad-4c40-8cad-e01bac8da826.png)
![2022-12-18 18_37_01-Virtua Cop - Elite Edition](https://user-images.githubusercontent.com/20983844/208336979-a5f2462a-5694-49e4-8ed8-75d2ae69ed91.png)

**Comment by seta-san** [comment]
guncon games almost certainly shouldn't be played with widescreen hacks. it's got to be throwing everything off.

**Comment by stenzek** [comment]
The default values are certainly incorrect in most cases. You'll need to tweak them.

It's really only a temporary measure, I expect using the screen offsets from the GS registers will be sufficient to line most games up automatically, but I didn't have the energy to do that in the initial PR.

**Comment by Gomam0n-afk** [comment]
> Virtua Cop: We're using the same game and it's perfect here, so it's related to how your calibrating/settings, not the emulator. It's flawless here, so try my steps again: ![2022-12-18 18_36_42-Virtua Cop - Elite Edition](https://user-images.githubusercontent.com/20983844/208336973-08cf3497-9aad-4c40-8cad-e01bac8da826.png) ![2022-12-18 18_37_01-Virtua Cop - Elite Edition](https://user-images.githubusercontent.com/20983844/208336979-a5f2462a-5694-49e4-8ed8-75d2ae69ed91.png)

Most settings are in default, my thinking is that you enabled something that made it work, not me.

**Comment by legend800** [comment]
nope i touched nothing, I just calibrated it properly in game and it works perfectly
![2022-12-18 19_15_46-PCSX2 Controller Settings](https://user-images.githubusercontent.com/20983844/208340602-e773ac02-e0bc-4f7e-82f9-486b9128e6dc.png)

**Comment by Gomam0n-afk** [comment]
![image](https://user-images.githubusercontent.com/59240947/208340886-698a8c9d-1f39-45fe-adcd-2f7015576858.png)
Same here, then what is it ? Magic ?
I was thinking something more related to the video side of things rather than the gcon2 settings itself.

**Comment by Gomam0n-afk** [comment]
here are my video settings, if anyone has an idea as to why this happens.
![image](https://user-images.githubusercontent.com/59240947/208341721-39139fb4-2ece-4614-98af-570db316b162.png)
![image](https://user-images.githubusercontent.com/59240947/208341741-ede817db-42a3-4857-a279-cb9e8d320ace.png)

**Comment by legend800** [comment]
Mine:
![2022-12-18 19_29_06-PCSX2 Settings](https://user-images.githubusercontent.com/20983844/208341926-fe875fe3-705b-42df-a275-f9d36199936b.png)

![2022-12-18 19_29_12-PCSX2 Settings](https://user-images.githubusercontent.com/20983844/208341929-d0c54d86-d370-49af-aeaf-0b31eaf7234b.png)

**Comment by kim-sama** [comment]
Hello. I tried this myself, I could only get Time Crisis 2 and 3 NTSC-U versions working. I tried these and they don't work:
Crisis Zone (PAL) SCES-52530
GunVari Collection + Time Crisis (NTSC-J) SLPS-25181
Time Crisis 3 (NTSC-J) SLPS-25290
Time Crisis II (NTSC-J) SLPS-20219
Vampire Night (NTSC-J) SLPS-25077
Virtua Cop Re Birth (NTSC-J) SLPS-62205

**Comment by IlDucci** [comment]
> Dino Stalker (SLES-50930) : BAD.

I was able to properly calibrate and beat the first stage of Dino Stalker PAL ENG. I just had to add manually the custom values used for the NTSC-U version:
`90,25 92,5 390 132 640 240`

**Comment by Comfarol** [comment]
You need to find the values of the games and provide them so they can be added here or add them if you know how. As you can see a lot of games are missing just because the values are unknown.

Also PAL games have more pixels so copying US configs that are there already wont always work.

https://github.com/PCSX2/pcsx2/blob/master/pcsx2/USB/usb-lightgun/guncon2.cpp#L67-L86

**Comment by dante3732** [comment]
Time Crisis jap work fine
Time Crisis 2 usa work fine
Time Crisis 3 usa work fine
Crisis Zone usa work fine
Dino Stalker usa work fine
Resident Evil Dead aim work fine
Gunfighter II europe work fine
Endgame usa work fine, but need plug guncon in usb 2

Virtua Cop pal calibration work good, but there is a very long delay before the shot is fired

Vampire Night usa can't calibrate
Resident Evil Gun Survivor 2 pal the mouse cursor does not match the sight in calibration

**Comment by legend800** [comment]
> Time Crisis jap work fine Time Crisis 2 usa work fine Time Crisis 3 usa work fine Crisis Zone usa work fine Dino Stalker usa work fine Resident Evil Dead usa aim work fine Gunfighter II europe work fine Endgame usa work fine, but need plug guncon in usb 2
> 
> Virtua Cop pal calibration work good, but there is a very long delay before the shot is fired
> 
> Vampire Night usa can't calibrate Resident Evil Gun Survivor 2 pal the mouse cursor does not match the sight in calibration

What are you settings for Time Crisis - Crisis Zone? Me and the other guy can't get past the Cali screen, even with the Cali shot button.

**Comment by dante3732** [comment]
> What are you settings for Time Crisis - Crisis Zone? Me and the other guy can't get past the Cali screen, even with the Cali shot button.

Default, i use cali buttons and all fine.

**Comment by dudesh34** [comment]
Wonder if they can find a way to disable the annoying flash effect...like some guy did on the old pcsx2 versions.

**Comment by dante3732** [comment]
> Wonder if they can find a way to disable the annoying flash effect...like some guy did on the old pcsx2 versions.

It seems that he had only for PS1?

**Comment by dante3732** [comment]
@Comfarol 90,25 102,75 422,50 137,00 638 228 it seems good for vampire night usa

**Comment by dudesh34** [comment]
> > Wonder if they can find a way to disable the annoying flash effect...like some guy did on the old pcsx2 versions.
> 
> It seems that he had only for PS1? https://forums.pcsx2.net/Thread-Nuvee-PS1-PS2-input-plugin-supporting-lightguns-by-Shalma

no, i mean to another guy that made it possible to play some ps2 titles without a flash...its working great for me on old pcsx2. here is the video that show it: https://www.youtube.com/watch?v=1N0nLXsw0Jg

**Comment by F0bes** [comment]
Looks like they replaced the texture, that's all. We support texture replacement, so you're free to do so yourself.
As I have said before (**in another issue**), this isn't something PCSX2 can simply provide out-of-box. People will have to patch each individual game. Please stop filling our issue tracker with your wants. If you want it done, learn how to do texture replacement and or game patching.

**Comment by legend800** [comment]
Yes, I've used that for years. It doesn't remove the fog at all, it just let's you play the game with a pad mapped to your mouse not guncon (no cali screen). Guncon mode causes the flash. Please don't spam the compat. tracker with conversation - take it to discord.

**Comment by F0bes** [comment]
> Yes, I've used that for years. It doesn't remove the fog at all, it just let's you play the game with a pad mapped to your mouse not guncon (no cali screen). Guncon mode causes the flash. Please don't spam the compat. tracker with conversation - take it to discord.

I don't believe you're right.

Firstly,
There are dumped textures and their replacement provided in the download link from the video.

Original
![image](https://user-images.githubusercontent.com/29295048/208546936-170ef2f0-a1ff-4f7c-af97-ee0d027eae90.png)
Replacement
![image](https://user-images.githubusercontent.com/29295048/208546954-3ec52bae-0ec5-405e-b1b0-d0c03ee34d84.png)

The above looks like what I'd expect if the goal was to remove a white filter on the screen.

Secondly,
![image](https://user-images.githubusercontent.com/29295048/208547113-0152f4b7-4137-4ffd-af90-ec15470cf67e.png)

The above is from the video linked. They are using a modified GSDX to replace the textures.

You're so quick to try and backseat moderate these trackers, you didn't even check to see what they posted.

**Comment by RedDevilus** [comment]
Oh that hacky texture replacements things again :s , there is a decent working on the nightly version.

![image](https://user-images.githubusercontent.com/24227051/208547610-aa4bcba1-4cf1-40c8-9741-c5d958bb1a70.png)

**Comment by legend800** [comment]
He referenced the "Mouse Patch" in the video. That's why I said that. Sorry I didn't download some random yt files to dive deeper.
![1](https://user-images.githubusercontent.com/20983844/208548171-f140e848-e138-4144-a948-72989b665ab4.png)

> >You're so quick to try and backseat moderate these trackers, you didn't even check to see what they posted.
Second, what are you even referencing? We're discussing GUNCON COMPATIBILITY and you are going off on some random texture replacement thing for a certain game. You do understand they are not connected? So instead of getting on my case, why don't you create a feature request for your unrelated thing or discuss on discord? That way, we can focus on guncon compatibility testing/results here.

**Comment by refractionpcsx2** [comment]
Closing in favour of #7618
